libtest/mock-transport.c

```diff
@@ -115,19 +134,28 @@ log_transport_mock_init(LogTransportMock *self, gchar *read_buffer1, gssize read
   while (buffer)
     {
       /* NOTE: our iov buffer is of a fixed size, increase if your test needs more chunks of data */
-      g_assert(self->iov_cnt < sizeof(self->iov) / sizeof(self->iov[0]));
-
-      if (length < 0)
-        length = strlen(buffer);
-
-      self->iov[self->iov_cnt].iov_base = buffer;
-      self->iov[self->iov_cnt].iov_len = length;
-      self->iov_cnt++;
+      g_assert(self->value_cnt < sizeof(self->value) / sizeof(self->value[0]));
+
+      if (length == -2)
```

Comment:
It is kinda hard to connect this `-2` value and the one in the `LTM_INJECT_ERROR` macro.
Would you create a constant for this ?
```c
#define LTM_INJECT_ERROR_LENGTH -2
#define LTM_INJECT_ERROR(err)   (GUINT_TO_POINTER(err)), LTM_INJECT_ERROR_LENGTH
```

Reply:
Done.

Reply:
Thank you, one small thing. Your update does not compile because you forget the `-2` from the `LTM_InJECT_ERROR_LENGTH` definition.